x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/rest/job/RestPostDataAction.java

```diff
@@ -27,15 +27,12 @@
 
     @Override
     public List<Route> routes() {
-        return Collections.emptyList();
-    }
-
-    @Override
-    public List<DeprecatedRoute> deprecatedRoutes() {
-        return Collections.singletonList(
-            new DeprecatedRoute(POST, MachineLearning.BASE_PATH + "anomaly_detectors/{" + Job.ID.getPreferredName() + "}/_data",
-            "Posting data directly to anomaly detection jobs is deprecated, " +
-                "in a future major version it will be compulsory to use a datafeed"));
+        return List.of(
+            Route.builder(POST, BASE_PATH + "anomaly_detectors/{" + Job.ID.getPreferredName() + "}/_data")
+                .deprecated("Posting data directly to anomaly detection jobs is deprecated, " +
+                    "in a future major version it will be compulsory to use a datafeed", RestApiCompatibleVersion.V_8
+                ).build()
+        );
```

Comment:
Same question about what the back port to 7.x will look like ?

Reply:
Marking this one as resolved, let's carry on the conversation in the other thread (https://github.com/elastic/elasticsearch/pull/69573#discussion_r586460388), though.